### PR TITLE
Fix 'sequence' lookup shortcut syntax and documentation

### LINKED ISSUE #22988: sequence lookup shortcut syntax doesn't work and wrong docs
##### ISSUE TYPE
 - Bug Report
 - Documentation Report

##### COMPONENT NAME
`sequence` lookup plugin

##### ANSIBLE VERSION
```
ansible 2.2.1.0
```
2.1, 2.3 and 2.4 have the same bug.

##### CONFIGURATION
Standard config

##### OS / ENVIRONMENT
N/A

##### SUMMARY
`with_sequence` shortcut syntax `[start-]end[/stride][:format]` is not honoured.

##### STEPS TO REPRODUCE
```
---
- hosts: localhost
  connection: local
  gather_facts: no
  tasks:
    - debug:
        msg: "{{ item }}"
      with_sequence: '5-6'
```

##### EXPECTED RESULTS
Loop over [5,6]:
```
TASK [debug] *******************************************************************
ok: [localhost] => (item=5) => {
    "item": "5",
    "msg": "5"
}
ok: [localhost] => (item=6) => {
    "item": "6",
    "msg": "6"
}
```

##### ACTUAL RESULTS
Parameter error:
```
TASK [debug] *******************************************************************
fatal: [localhost]: FAILED! => {"failed": true, "msg": "unknown error parsing with_sequence arguments: u'5-6'. Error was: unrecognized arguments to with_sequence: [u'_raw_params']"}
```

Also there is wrong parameters passing in the docs:
```
    - user:
        name: "{{ item }}"
        state: present
        groups: "evens"
      with_sequence:
        - start: 0
        - end: 32
        - format: testuser%02x
```

Parameters can be passed to `with_sequence` only as string, not as list or dict.

### PR DESCRIPTION
**Pull request by berlic**
##### SUMMARY
Fixes #22988

Added missing `return` statement in `parse_simple_args`.  
Plugin always tried to parse named arguments without it as `parse_simple_args` never returned `True`.

Also corrected misleading documentation, because sequence plugin doesn't accept parameters as dictionary, only as key=value string.

##### ISSUE TYPE
 - Bugfix Pull Request
 - Docs Pull Request

##### COMPONENT NAME
`sequence` plugin

##### ANSIBLE VERSION
```
ansible 2.4.0 (sequence-lookup-fix 76820cf71d) last updated 2017/03/27 07:39:56 (GMT +000)
  config file =
  configured module search path = Default w/o overrides
  python version = 2.7.12 (default, Nov 19 2016, 06:48:10) [GCC 5.4.0 20160609]
```

##### ADDITIONAL INFORMATION
This should be ported to all 2.x releases (2.1, 2.2, 2.3). I don't know how to do it correctly.

### CODE REVIEW

**Review by dharmabumstead — APPROVED** [review]
Docs OK with a few edits. Needs tech review from the core team. Thanks!

File: docs/docsite/rst/playbooks_loops.rst
Comment (bcoca) [inline docs/docsite/rst/playbooks_loops.rst:283]: this is not really true, this is one of 2 options, the YAML format is also acceptable.

File: docs/docsite/rst/playbooks_loops.rst
Comment (bcoca) [inline docs/docsite/rst/playbooks_loops.rst:308]: ~~no need to convert all examples to k=v, we can leave one of each style at least~~

**Review by bcoca — CHANGES_REQUESTED** [review]

File: docs/docsite/rst/playbooks_loops.rst
Comment (berlic) [reply docs/docsite/rst/playbooks_loops.rst:283]: @bcoca actually, no, you can't use YAML format. I double checked it.

in case of:
```
      with_sequence:
        - count: 4
```
the error is
> unknown error parsing with_sequence arguments: {u'count': 4}. Error was: expected string or buffer

in case of:
```
      with_sequence:
        count: 4
```
the error is
> unrecognized arguments to with_sequence: [u'_raw_params']

I tried to add easy fix to achieve list-dict syntax like in the docs by doing:
```
                    if isinstance(term, dict):
                        self.parse_kv_args(term) # don't parse k=v if term is a dict
                    else:
                        if not self.parse_simple_args(term):
                            self.parse_kv_args(parse_kv(term))
```
but this still requires to rewrite `parse_kv_args` because of
> Error was: int() can't convert non-string with explicit base

There's open issue #13879 about this.

**Comment by bcoca** [comment]
so this seems to be from the 'blanket' standardization 96538ec15803744912f60d472c03b9049d3cb4a5

we probably have many unworkable examples

File: docs/docsite/rst/playbooks_loops.rst
Comment (bcoca) [reply docs/docsite/rst/playbooks_loops.rst:283]: looked at code, you are correct, this was victim of 'blanket' yamliziation of examlpes

**Comment by berlic** [comment]
Is there anything else I can do about this PR?

**Review by bcoca — APPROVED** [review]

**Pull request merged**
